File: src/JavaScript/Algorithms/math/fibonacci/fibonacci.js
function fibonacci(n) {
    const sequence = [1];

    let currentValue = 1;
    let previousValue = 0;
    
    if (n === 1) {
        return sequence;
    }

    let iterationCounter = n - 1;

    while (iterationCounter) {
        currentValue += previousValue;
        previousValue = currentValue - previousValue;

        sequence.push(currentValue);

        iterationCounter -= 1;
    }
    
    return sequence;
}

File: src/Python/Algorithms/math/fibonacci/fibonacci.py
def fibonacci(n):
    sequence = [1]

    currentValue = 1
    previousValue = 0

    if (n == 1):
        return sequence
    
    iterationCounter = n - 1

    while (iterationCounter):
        currentValue += previousValue
        previousValue = currentValue - previousValue

        sequence.append(currentValue)

        iterationCounter -= 1

    return sequence